hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -1359,6 +1365,7 @@ public InternalDirOfViewFs(final InodeTree.INodeDir<FileSystem> dir,
       showMountLinksAsSymlinks = config
           .getBoolean(CONFIG_VIEWFS_MOUNT_LINKS_AS_SYMLINKS,
               CONFIG_VIEWFS_MOUNT_LINKS_AS_SYMLINKS_DEFAULT);
+      this.config = config;
```

Comment:
> else the ugi logic...

Yeah and we can set mountLinkGroupName & mountLinkUserName for the ugi logic so that we can retrieve the info directly next time we call the getters.